namerd/storage/in-memory/src/main/scala/io/buoyant/namerd/storage/InMemoryDtabStore.scala

```diff
@@ -31,8 +31,11 @@ class InMemoryDtabStore(namespaces: Map[String, Dtab]) extends DtabStore {
       }
     }
 
-  def list(): Future[Set[String]] =
-    Future.value(dtabStatesMu.synchronized(dtabStates).keySet)
+  def list(): Future[Set[String]] = Future.value {
+    dtabStatesMu.synchronized(dtabStates).filter {
+      case (key, value) => value.sample.isDefined
+    }.keySet
+  }
```

Comment:
tioli:

I generally prefer to avoid nesting -- easier to understand:
```
val v = ..
Future.value(v)
```

might write this as:
```scala
def list(): Future[Set[String]] = {
  val keys = dtabStatesMu.synchronized(dtabStates).collect {
    case (k, v) if v.sample.isDefined => k
  }
  Future.value(keys.toSet)
}
```